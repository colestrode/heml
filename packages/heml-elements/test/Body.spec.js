const render = require('@heml/render')
const Body = require('../build/Body').default
const Style = require('../build/Style').default
const Meta = require('../build/Meta').default

beforeAll(() => {
  Style.preRender({ elements: [] });
  Meta.preRender();
})

test('renders with some attrs', () => {
  const attrs = {
    margin: '1px 2px 3px 4px'
  }

  return render.renderElement(Body, attrs, 'Lady Dynamite')
    .then((res) => {
      expect(res).toMatchSnapshot();
    });
});
