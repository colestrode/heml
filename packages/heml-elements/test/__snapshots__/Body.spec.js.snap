// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with some attrs 1`] = `"<body margin=\\"1px 2px 3px 4px\\" class=\\"body\\" style=\\"margin: 0; width: 100%;\\"><table class=\\"bodyTable\\" role=\\"presentation\\" width=\\"100%\\" align=\\"left\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"margin: 0;\\"><tr><td class=\\"body__content\\" align=\\"left\\" width=\\"100%\\" valign=\\"top\\">Lady Dynamite</td></tr></table><div style=\\"display:none; white-space:nowrap; font-size:15px; line-height:0;\\">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </div></body>"`;
